app/src/main/java/com/improve/latetrain/AddMinsFragment.kt

```diff
@@ -1,102 +1,254 @@
- * Activities that contain this fragment must implement the
- * [AddMinsFragment.OnFragmentInteractionListener] interface
- * to handle interaction events.
- * Use the [AddMinsFragment.newInstance] factory method to
- * create an instance of this fragment.
- *
- */
 class AddMinsFragment : Fragment() {
-    // TODO: Rename and change types of parameters
-    private var param1: String? = null
-    private var param2: String? = null
-//    private var listener: OnFragmentInteractionListener? = null
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        arguments?.let {
-            param1 = it.getString(ARG_PARAM1)
-            param2 = it.getString(ARG_PARAM2)
+    private val TAG = "ADD_MIN_FRAG_TAG"
+    private val LAST_CLICK = "LAST_CLICK"
+    private val LOCATION_PERMISSION_GIVEN = "LOCATION_PERMISSION_GIVEN"
+    private val REQUEST_CHECK_SETTINGS = 0x1
+    private val MY_PERMISSIONS_REQUEST_LOCATION = 0x2
+    private val IS_PERMISSION_REQUEST_GRANTED = "IS_PERMISSION_REQUEST_GRANTED"
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?):
+            View? = inflater.inflate(R.layout.fragment_add_mins, container, false)
+
+
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        (animation_layout as LottieAnimationView).playAnimation()
+
+        activity?.let {
+            sharedPreferences = it.getSharedPreferences("com.improve.latetrain", Context.MODE_PRIVATE)
         }
-    }
 
-    override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View? {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_add_mins, container, false)
-    }
+        addMinBtn.setOnClickListener {
+            val lastTime = sharedPreferences.getLong(LAST_CLICK, 0)
+            Log.d(TAG, lastTime.toString())
+            var is30MinPass = true
+            if (lastTime + 1800 > System.currentTimeMillis() / 1000) {
+                is30MinPass = false
+                val builder = AlertDialog.Builder(context)
+                builder.setTitle(getString(R.string.wait_addmins))
+                builder.setMessage(getString(R.string.once_in_30_addmins))
+                builder.setPositiveButton(getString(R.string.got_it_addmins)) { _, _ -> }
+                val dialog: AlertDialog = builder.create()
+                dialog.show()
+                return@setOnClickListener
+            }
 
-    // TODO: Rename method, update argument and hook method into UI event
-    fun onButtonPressed(uri: Uri) {
-//        listener?.onFragmentInteraction(uri)
-    }
+            var minutes = 0
+            if (minLateEt.selectedItem.toString() != "") {
+                minutes = minLateEt.selectedItem.toString().toInt()
```

Comment:
lets discuss this later, this is not a blocker, but remind me to talk about how to parse string to int in better ways.